modules/twinklespeedy.js

```diff
@@ -397,6 +397,14 @@ Twinkle.speedy.callback.modeChanged = function twinklespeedyCallbackModeChanged(
 			work_area.append( { type: 'header', label: 'User pages' } );
 			work_area.append( { type: radioOrCheckbox, name: 'csd', list: Twinkle.speedy.generateCsdList(Twinkle.speedy.userAllList, mode) } );
 		}
+		else if (namespace == 0) {
+			work_area.append( { type: 'header', label: 'Mainspace redirects' } );
+			work_area.append( { type: radioOrCheckbox, name: 'csd', list: Twinkle.speedy.generateCsdList(Twinkle.speedy.mainspaceRedirect, mode) } );
+		}
+		else if (namespace == 6) {
+			work_area.append( { type: 'header', label: 'Mainspace redirects' } );
```

Comment:
Mooted by my comment on the PR, but this is mislabeled

Reply:
> Mooted by my comment on the PR, but this is mislabeled

Oh, it should be file redirects. How should I ensure that there isn't a heading? Remove lines 401 and 405?

Reply:
You'd need to refactor it to use `concat`, like we do with `userNonRedirectList` (L362) or `generalNonUser` (L415).  Basically, take `redirectList`, then:
- if it's mainspace throw `articleRedirectList` in front
- if it's filespace throw `fileRedirectList` in back
- Append that to the `Redirects` section

Reply:
Done in 2 more commits, now uses concat.